tests/php/ORM/DataObjectTest.php

```diff
@@ -2671,4 +2672,48 @@ public function testDBObjectEnum()
         $vals = ['25.25', '50.00', '75.00', '100.50'];
         $this->assertSame(array_combine($vals ?? [], $vals ?? []), $obj->dbObject('MyEnumWithDots')->enumValues());
     }
+
+    public function provideTestGetDatabaseBackedField()
+    {
+        return [
+            ['Captain.IsRetired', 'Captain.IsRetired'],
+            ['Captain.ShirtNumber', 'Captain.ShirtNumber'],
+            ['Captain.FavouriteTeam', null],
+            ['Captain.FavouriteTeam.Fans', null],
+            ['Captain.FavouriteTeam.Fans.Count', null],
+            ['Captain.FavouriteTeam.Title', 'Captain.FavouriteTeam.Title'],
+            ['Captain.FavouriteTeam.Title.Plain', 'Captain.FavouriteTeam.Title'],
+            ['Captain.FavouriteTeam.ReturnsNull', null],
+            ['Captain.FavouriteTeam.MethodDoesNotExist', null],
+            ['Captain.ReturnsNull', null],
+            ['Founder.FavouriteTeam.Captain.ShirtNumber', 'Founder.FavouriteTeam.Captain.ShirtNumber'],
+            ['Founder.FavouriteTeam.Captain.Fans', null],
+            ['Founder.FavouriteTeam.Captain.Fans.Name.Plain', 'Founder.FavouriteTeam.Captain.Fans.Name'],
+            ['Founder.FavouriteTeam.Captain.ReturnsNull', null],
+            ['HasOneRelationship.FavouriteTeam.MyTitle', null],
+            ['SubTeams.Comments.Name.Plain', 'SubTeams.Comments.Name'],
+            ['Title', 'Title'],
+            ['Title.Plain', 'Title'],
+            ['DatabaseField', 'DatabaseField'],
+            ['DatabaseField.MethodDoesNotExist', 'DatabaseField'],
+            ['ReturnsNull', null],
+            ['DynamicField', null],
+            ['SubTeams.ParentTeam.Fans', null],
+            ['SubTeams.ParentTeam.Founder.FoundingTeams', null],
```

Comment:
These two cases to cover `array_shift` in case `if ($component instanceof Relation || $component instanceof DataList)`.